libs/types-common/src/Data/Code.hs

```diff
@@ -35,14 +35,17 @@ import Data.Scientific (toBoundedInteger)
 import Data.Text.Ascii
 import Data.Time.Clock
 import Imports
+import Test.QuickCheck (Arbitrary (arbitrary))
 
 -- | A scoped identifier for a 'Value' with an associated 'Timeout'.
 newtype Key = Key {asciiKey :: Range 20 20 AsciiBase64Url}
-  deriving (Eq, Show, FromJSON, ToJSON, FromByteString, ToByteString)
+  deriving (Eq, Show)
+  deriving newtype (FromJSON, ToJSON, FromByteString, ToByteString, Arbitrary)
```

Comment:
But it doesn't change the instance. Which deriving strategy do you think it used before? It will only use anyclass if `LANGUAGE DeriveAnyClass` is enabled, which it isn't (and even then it would fail, since the default instance uses `Generic`, of which there is no instance).

So I think it didn't change the behaviour, however, I've gotten confused about this a few times before, so I might be missing something. This confusion is exactly the reason for using explicit deriving strategies. We should always do it and enforce this (which brings us back to using hlint).

Reply:
Ok you are right:

```
Prelude Data.Aeson> newtype Y = Y Int deriving newtype (Show, ToJSON)
newtype Y = ...
Prelude Data.Aeson> newtype X = X Int deriving (Show, ToJSON)
newtype X = ...
Prelude Data.Aeson> (X 3, Y 3, encode (X 3, Y 3))
(X 3,3,"[3,3]")  -- (same for ToByteString)
```

I earlier tested the `Show` instance, and there is a difference, but that's why you left this in `deriving stock`.